### PR TITLE
Handle duplicate XAML files in codegen

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes https://github.com/unoplatform/private/issues/433

### CODE REVIEW

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.2

File: src/SourceGenerators/SourceGeneratorHelpers/Helpers/GeneratorExecutionContextExtensions.cs [hunk 1]
```diff
@@ -67,5 +68,17 @@ public string GetMetadataValue(string name)
...
 			return string.IsNullOrEmpty(metadataValue) ? "" : metadataValue!;
 		}
+
+		public static IEqualityComparer<MSBuildItem> IdentityComparer
+			=> _identityComparer ??= new MSBuildItemIdentityComparer();
+
+		private class MSBuildItemIdentityComparer : IEqualityComparer<MSBuildItem>
+		{
+			public bool Equals(MSBuildItem x, MSBuildItem y)
+				=> string.Equals(x.Identity, y.Identity, System.StringComparison.OrdinalIgnoreCase);
```
Comment (Youssef1313) [inline src/SourceGenerators/SourceGeneratorHelpers/Helpers/GeneratorExecutionContextExtensions.cs:78]: Not sure if we should use StringComparison based on file system's case sensitivity?

**Review by Youssef1313 — APPROVED** [review]

File: src/SourceGenerators/SourceGeneratorHelpers/Helpers/GeneratorExecutionContextExtensions.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SourceGenerators/SourceGeneratorHelpers/Helpers/GeneratorExecutionContextExtensions.cs:78]: I wondered, the problem is that it's not easy to determine if the file system should be case sensitive, as it depends on the file location. In our case, having case sensitivity differences is likely to cause other bugs.

**Review by agneszitte — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs [hunk 2]
```diff
@@ -172,8 +172,9 @@ public XamlCodeGeneration(GeneratorExecutionContext context)
 			var applicationDefinitionItems = GetWinUIItems("ApplicationDefinition").Concat(GetWinUIItems("UnoApplicationDefinition"));
 
 			var xamlItems = pageItems
+				.Except(applicationDefinitionItems, MSBuildItem.IdentityComparer)
 				.Concat(applicationDefinitionItems)
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs:176]: Except(applicationDefinitionItems).Concat(applicationDefinitionItems) is quite confusing 😕

Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs:176]: It's the order, I want to keep the original itemgroup, name favoring the `ApplicationDefinition` over the `Page.

**Pull request merged**